### ISSUE TITLE
[feat] Add right_shift_logits (Shift Operation) to DiffusionTrainer for Dream fine-tuning

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle#201]
## Background

Dream 7B uses an AR model (Qwen 2.5) as initialization and applies a **Shift Operation** to maintain positional alignment between the AR pre-training objective and the diffusion masking objective during fine-tuning.

Specifically, when a token at position `i+1` is masked, the model applies the logit from position `i` to predict the token at `i+1`, mirroring how AR models shift predictions by one. This allows full knowledge transfer from the AR pre-trained weights without catastrophic forgetting.

### Reference implementations
- `dev/repos/dllm/dllm/core/trainers/mdlm.py`: `MDLMConfig.right_shift_logits` (bool), `_preprocess_inputs` / `_postprocess_outputs`
- `dev/repos/Dream/src/trainer/fsdp_sft_trainer.py` L777–790: `shift_logits = torch.cat([logits[:, 0:1], logits[:, :-1]], dim=1)`
- `unturtle/models/dream/generation_utils.py` L564–569: shift already applied at inference time

### Current state
`DiffusionTrainer` does **not** have a `right_shift_logits` option. Users fine-tuning Dream with `DiffusionTrainer` will compute loss against unshifted logits, which is incorrect for Dream weights.

The inference side (`DreamGenerationMixin`) already applies the shift correctly, so the gap is only on the training side.

## Acceptance criteria

- [ ] Add `right_shift_logits: bool = False` to `DiffusionTrainingArguments`
- [ ] In `DiffusionTrainer.compute_loss`, when `right_shift_logits=True`, apply the logit shift before loss computation:
  ```python
  # shift: logit[i] predicts position i+1
  logits = torch.cat([logits[:, :1], logits[:, :-1]], dim=1)
  ```
- [ ] Document that this must be `True` when fine-tuning Dream checkpoints
- [ ] Add test: fine-tuning with a Dream-like model, verify loss decreases when `right_shift_logits=True`

## Labels
type: feat, diffusion
